### PR TITLE
Split synapse neuron models

### PR DESCRIPTION
**Pull request by rowleya**
Makes it possible to 
1) Make a region "referenceable" (i.e. assign it an ID that can be used later)
2) Reference a region with an ID.

The region to be referenced must be on the same chip as the region that makes the reference.  There is also a context object to make it easier to store and retrieve references in, well, a frame of reference.

This is needed by the split synapse-neuron model code to make it possible to generate the synaptic data once and then reference it from all the synapse cores.

### CODE REVIEW

File: data_specification/memory_region.py [hunk 1]
```diff
@@ -15,99 +15,5 @@
 
 
 class MemoryRegion(object):
-    """ Memory region storage object.
+    """ Identifies something as a Memory region
```
Comment (andrewgait) [inline data_specification/memory_region.py:18]: Seems to me like this should be renamed to AbstractMemoryRegion if it is just going to be an empty base class for the Real and Reference classes?

**Review by andrewgait — CHANGES_REQUESTED** [review]
Looks good generally; just a minor comment on naming.

File: data_specification/memory_region.py (hunk 1 above)
Comment (rowleya) [reply data_specification/memory_region.py:18]: Makes sense - not sure why I didn't as that also solves the problem of having things reference the old version of this!  I will rename...

Comment (rowleya) [reply data_specification/memory_region.py]: Fixed

**Review by andrewgait — APPROVED** [review]

**Pull request merged**